### PR TITLE
21w41a server world

### LINKED ISSUE #2755: ChunkTicketManager.getSpawningChunkCount -> getTickedChunkCount
"Spawning chunk" as a terminology to refer to 128-block-radius-chunk is confusing:

- similar phrase, "spawn chunk", is used to refer to the chunks around world spawn point
- not all mobs use 128-block-radius spawning radius
- actual spawning is based on a 3D sphere, not 2D circle

"Ticked chunk" is less ambiguous, and is technically true (chunks, and therefore blocks, only get ticked within that radius)

### PR DESCRIPTION
**Pull request by apple502j**
Resolves #2755

### CODE REVIEW

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 1]
```diff
@@ -151,14 +154,61 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
...
 		ARG 1 chunkPos
-	METHOD method_31413 addEntitiesFromNbt (Lnet/minecraft/class_3218;Ljava/util/List;)V
+	METHOD method_27053 mark (Lnet/minecraft/class_1923;Lnet/minecraft/class_2806$class_2808;)B
+		ARG 1 pos
+		ARG 2 type
+	METHOD method_27054 markAsProtoChunk (Lnet/minecraft/class_1923;)V
+		ARG 1 pos
+	METHOD method_27055 isLevelChunk (Lnet/minecraft/class_1923;)Z
+		ARG 1 pos
+  	METHOD method_31413 addEntitiesFromNbt (Lnet/minecraft/class_3218;Ljava/util/List;)V
```
Comment (enbrain) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]:
```suggestion
	METHOD method_31413 addEntitiesFromNbt (Lnet/minecraft/class_3218;Ljava/util/List;)V
```

**Review by enbrain — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 2]
```diff
@@ -28,6 +28,9 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
 	FIELD field_19343 unloadTaskQueue Ljava/util/Queue;
 	FIELD field_23786 chunkToType Lit/unimi/dsi/fastutil/longs/Long2ByteMap;
 	FIELD field_26931 chunkStatusChangeListener Lnet/minecraft/class_5567;
+	FIELD field_29671 PROTOCHUNK B
```
Comment (Juuxel) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]:
```suggestion
	FIELD field_29671 PROTO_CHUNK B
```

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 3]
```diff
@@ -28,6 +28,9 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
...
+	FIELD field_29672 UNMARKED_CHUNK B
+	FIELD field_29673 LEVELCHUNK B
```
Comment (Juuxel) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]:
```suggestion
	FIELD field_29673 LEVEL_CHUNK B
```

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by Juuxel — APPROVED** [review]

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 4]
```diff
@@ -159,6 +169,46 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
...
+		ARG 0 chunkPos
+		ARG 1 sectionX
+		ARG 2 sectionZ
+		ARG 3 distance
+	METHOD method_37901 isWithinDistance (Lnet/minecraft/class_1923;Lnet/minecraft/class_3222;ZI)Z
+		ARG 0 chunkPos
+		ARG 1 player
+		ARG 2 usePlayerWatchedSection
+		ARG 3 distance
+	METHOD method_37905 isInDistanceEdge (IIIII)Z
```
Comment (enbrain) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]: Is this whether the position is on a circle’s circumference? If so,
```suggestion
	METHOD method_37905 isOnDistanceEdge (IIIII)Z
```

I'd probably name it `isOnBorder` but `distanceEdge` is already used I guess.

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 5]
```diff
@@ -159,6 +169,46 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
...
+		ARG 0 x1
+		ARG 1 z1
+		ARG 2 x2
+		ARG 3 z2
+		ARG 4 distance
+	METHOD method_37906 isInDistanceEdge (Lnet/minecraft/class_1923;Lnet/minecraft/class_3222;ZI)Z
```
Comment (enbrain) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]:
```suggestion
	METHOD method_37906 isOnDistanceEdge (Lnet/minecraft/class_1923;Lnet/minecraft/class_3222;ZI)Z
```

**Review by enbrain — APPROVED** [review]

**Pull request merged**